keyhive_core/src/transact.rs

```diff
@@ -254,17 +254,17 @@ pub async fn transact_nonblocking<
 /// ```
 #[instrument(skip_all)]
 pub async fn transact_async<
-    T: MergeAsync + Clone,
+    T: MergeAsync,
     Error,
     F: AsyncFnOnce(T::AsyncForked) -> Result<T::AsyncForked, Error>,
 >(
-    trunk: &T,
+    trunk: &mut T,
     tx: F,
 ) -> Result<(), Error> {
     let forked = trunk.fork_async().await;
     let diverged = info_span!("async_transaction")
         .in_scope(|| async { tx(forked).await })
         .await?;
-    trunk.clone().merge_async(diverged).await;
+    trunk.merge_async(diverged).await;
```

Comment:
@alexjg the idea here was to not have unique ownership of the trunk by the transaction, so you need to wrap it in something like a `Rc<RefCell<_>>`

Reply:
Yep this was 100% a hack, this is what I put in the commit message:

> For now, I have removed the `Send` bound and 
the `Clone` bound, but this obviously makes `MergeAsync` less useful.
However, right now we only use it in places where we don't need `Send`
futures (i.e. we're running on a local runtime), so we can come back to
this later.

Reply:
Ah okay! Sorry multitasking at the moment, and should stop jumping the gun as it were